spacy/lang/lex_attrs.py

```
@@ -134,21 +171,60 @@ def word_shape(text):
+    return string.isspace()
+
+
+def is_title(string):
+    return string.istitle()
+
+
+def is_upper(string):
+    return string.isupper()
+
+
+def is_stop(string, stops=set()):
```

Ruff (0.12.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)